docs/IsolationNetworkPolicies.md

```diff
@@ -0,0 +1,124 @@
+of IPs. These extensions will not break CRDs backward compatability.
+
+The `Selector` will have the following structure: 
+
+```go
+type Selector struct {
+	// Cluster name 
+	//+optional 
+	ClusterName string `json:"clusterName"`
+	
+	WorkloadSelector enables to connect the resource to the application 
+	// Applications labels should match the labels in the selector. 
+	//+required 
+	WorkloadSelector metav1.LabelSelector `json:"workloadSelector"`
+	
+	// Namespaces where user application might run
+	// +optional 
+	Namespaces []string `json:"namespaces"`
+
+	// IPBlocks define policy on particular IPBlocks.
+	// the structure of the IPBlock is defined at https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#ipblock-v1-networking-k8s-io
+	// +optional
+	IPBlocks []*IPBlock `json:"ipBlocks,omitempty"`
+}
+```
+The mapping the `Selector` elements into the `from` NP ingress definitions will be done according the following algorithm:
+- if `Namespaces` is empty, a [NetworkPolicyPeer](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#networkpolicypeer-v1-networking-k8s-io)
+  with `podSelector` equals to `WorkloadSelector` will be created. Which means that only pods with the defined labels from 
+  the module namespace will be able to connect, or be connected.
+- If `Namespaces` is not empty, for each entry in the array will be created a `NetworkPolicyPeer`
+   with `podSelector` equals to `WorkloadSelector` and a `namespaceSelector` contains a single `kubernetes.io/metadata.name`
+   label equals to the entry from `Namespaces`. [Here](https://kubernetes.io/docs/concepts/services-networking/network-policies/#targeting-a-namespace-by-its-name) 
+   you can read more about targeting a Namespace by its name. If `Namespaces` is not empty but `WorkloadSelector` is empty, 
+   then any pod from the selected namespaces will be able to connect.
+
+- If `IPBlocks` is not empty, a separate NetworkPolicyPeer will be created for each IPBlocks entry.
+
+### FybrikModule extensions
+
+Some of Fybrik modules need external services for their functionality. For example, a currency exchange module checks 
+the currency rates form an external service.
+We suggest adding an `externalServices` array into the `FybrikModule.Spec`. Each entry will be a DNS name or IP address of 
+the external service. During `egress` NetworkPolicyPeer creations, each service name will be translated to a single IPBlock entry.
+
+### PlotterController extensions
+
+In order to create ingress policies for modules, the blueprint that creates NPs should know potential clients, therefore
+PlotterController who has the entire picture, should provide this information. 
+
+NetworkPolicies select pods according to labels. In order to separate policies for 2 different modules, each module should
+have a unique label, in additional to the common, fybrik application defined labels.
```

Comment:
Fybrik passes a set of labels inside Helm values but does not enforce these labels on the deployed resources. No one guarantees that a 3rd party module will propagate the labels to its pods.